source/reference/headings.txt

```diff
@@ -0,0 +1,155 @@
+.. _style-guide-headings:
+
+========
+Headings
+========
+
+.. default-domain:: mongodb
+
+Use over and underlines consisting of a repeating character to specify a heading
+in rST.
+
+Heading Level Markup
+--------------------
+
+The character used for the over and underlines has no effect on the heading
+level, as long as the order used is consistent. However, to maintain consistency
+and readability throughout the documentation, use the following heading level
+structure:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 10 30 10 40 10
+
+   * - Heading Level
+     - Character
+     - Under or Overline
+     - Common Usage
+     - Renders in HTML as
+
+   * - 1
+     - ``=`` (Equals sign)
+     - Both
+     - Page Titles Only
+     - ``<h1>``
+
+   * - 2
+     - ``-`` (Hyphen)
+     - Under
+     - Main Headings
+     - ``<h2>``
+
+   * - 3
+     - ``~`` (Tilde)
+     - Under
+     - Subheadings
+     - ``<h3>``
+
+   * - 4
+     - ````` (Backtick)
+     - Under
+     - 2nd Level Subheadings
+     - ``<h4>``
+
+   * - 5
+     - ``+`` (Plus sign)
+     - Under
+     - 3rd Level Subheadings
+     - ``<h5>``
+
+Heading Markup Rules
+--------------------
+
+Headings must follow these rules to prevent build errors:
+
+- Make the underline at least as long as the text above it.
+
+  .. list-table::
+     :header-rows: 1
+     :widths: 50 50
+
+     * - Use
+       - Avoid
+
+     * -
+
+         .. code-block:: rst
+            :copyable: false
+
+            Create One New User
+            -------------------
+
+       -
+
+         .. code-block:: rst
+            :copyable: false
+
+            Create One New User
+            --------------
+
+- When using both over and underlines, their lengths must be identical.
+
+  .. list-table::
+     :header-rows: 1
+     :widths: 50 50
+
+     * - Use
+       - Avoid
+
+     * -
+
+         .. code-block:: rst
+            :copyable: false
+
+            -------------------
+            Create One New User
+            -------------------
+
+       -
+
+         .. code-block:: rst
+            :copyable: false
+
+            -------------
+            Create One New User
+            -------------------
+
+- Don't skip Heading Levels. Heading Level 3 cannot come after
+  Heading Level 1.
+
+  .. list-table::
+     :header-rows: 1
+     :widths: 50 50
+
+     * - Use
+       - Avoid
+
+     * -
+
+         .. code-block:: rst
+            :copyable: false
+
+            ============
+            Manage Users
+            ============
+
+            Create One New User
+            -------------------
+
+       -
+
+         .. code-block:: rst
+            :copyable: false
+
+            ============
+            Manage Users
+            ============
+
+            Create One New User
+            ~~~~~~~~~~~~~~~~~~~
+
+- Avoid using {+rst+} replacements-such as source constants-in heading text. If the
```

Comment:
```suggestion
- Avoid using {+rst+} replacements, such as :ref:`source constants <source-constants>`, in heading text. If the
```
Suggestion: avoid using hyphenated asides -- they aren't very common in our docs language, so they seem out of place. Link to source constants since readers may not know what a source constant is, and this bullet point doesn't make much sense without that knowledge.